forta-bots/Debt-Ceiling-Utilization/src/agent.ts

```diff
@@ -3,23 +3,46 @@ import { providers, BigNumber } from "ethers";
 import NetworkData, { NETWORK_MAP } from "./network";
 import NetworkManager from "./network";
 import Fetcher from "./fetcher";
-import { createFinding, UTILIZATION_THRESHOLD } from "./utils";
+import { createFinding, FILE_IFACE, UTILIZATION_THRESHOLD } from "./utils";
 
 const networkManager: NetworkData = new NetworkManager(NETWORK_MAP);
+let router: { domains: string[] } = {
+  domains: [],
+};
 
-export const initialize =
-  (provider: providers.Provider): Initialize =>
+export const provideInitialize =
+  (router: { domains: string[] }, provider: providers.Provider, data: NetworkData, fetcher: Fetcher): Initialize =>
   async () => {
     const { chainId } = await provider.getNetwork();
     networkManager.setNetwork(chainId);
+
+    router.domains = await fetcher.getDomains(data.TeleportRouter, await provider.getBlockNumber());
```

Comment:
needs to be moved to the handler logic, so the `init` flag will be needed as in other bots.